synapse/storage/databases/main/account_data.py

```diff
@@ -158,10 +158,34 @@ def get_account_data_for_user_txn(
             txn.execute(sql, (user_id,))
             rows = self.db_pool.cursor_to_dict(txn)
 
-            global_account_data = {
+            return {
                 row["account_data_type"]: db_to_json(row["content"]) for row in rows
             }
 
+        return await self.db_pool.runInteraction(
+            "get_global_account_data_for_user", get_global_account_data_for_user
+        )
+
+    @cached()
+    async def get_room_account_data_for_user(
+        self, user_id: str
+    ) -> Dict[str, Dict[str, JsonDict]]:
```

Comment:
Can we return `Mapping`s here please, so mypy can stop us from accidentally mutating the cache's contents?

Reply:
And the same for get_global_account_data_for_user above.